catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/SecureHadoopCatalogOperations.java

```diff
@@ -262,17 +265,21 @@ public void testConnection(
   @Override
   public List<PathContext> getPathContext(NameIdentifier filesetIdentifier) {
     Fileset fileset = loadFileset(filesetIdentifier);
-    Map<String, String> locations = fileset.storageLocations();
-    Preconditions.checkArgument(
-        locations != null && !locations.isEmpty(),
-        "No storage locations found for fileset: " + filesetIdentifier);
-
-    // todo: support multiple storage locations
-    String path = locations.get(fileset.properties().get(PROPERTY_DEFAULT_LOCATION_NAME));
+    CallerContext callerContext = CallerContext.CallerContextHolder.get();
+    String targetLocationName =
+        callerContext != null
+            ? callerContext.context().get(CredentialConstants.HTTP_HEADER_CURRENT_LOCATION_NAME)
+            : fileset.properties().get(PROPERTY_DEFAULT_LOCATION_NAME);
+    String path = fileset.storageLocations().get(targetLocationName);
```

Comment:
Could you use a method to get the location from a fileset and an optional location name?